libbeat/common/mapstr_pointer.go

```diff
@@ -0,0 +1,29 @@
+package common
+
+import (
+	"sync/atomic"
+	"unsafe"
+)
+
+// MapStrPointer stores a pointer to atomically get/set a MapStr object
```

Comment:
I would have prefered to have "optimizations" like this in a follow up PR to have the code we put in as simple as possible to start with and then when we do optimizations we also add some benchmark tests for it. Makes it easier to later detect in case these optimizations causes some issues, if it happened with the optimization or the original code.

Reply:
Agreed, but as I try to get a stable API to introduce new providers as soon as possible this somehow made sense while I waited for reviews. Once this is merged, autodiscover API should have everything we need